### PR TITLE
Add phpstan and resolve errors

### PR DESCRIPTION
**Pull request by hosni**

### CODE REVIEW

File: src/Http/Controllers/GraphqlController.php [hunk 1]
```diff
@@ -44,15 +55,24 @@ public function run(Request $request)
...
 				$directives = $info->fieldDefinition->astNode->directives;
 				$count = $directives->count();
 
 				$controller = $method = null;
 				for ($i = 0; $i < $count; $i++) {
 					$node = $directives->offsetGet($i);
-					if ($node->name->value == "method") {
-						[$controller, $method] = explode("@", $node->arguments->offsetGet(0)->value->value, 2);
-						break;
+					/** check existence of value,
```
Comment (yeganemehr) [inline src/Http/Controllers/GraphqlController.php]: this is phpdoc not comment

File: composer.json [hunk 2]
```diff
@@ -10,6 +11,16 @@
         "jalno/lumen": "^0.0.1",
         "webonyx/graphql-php": "^14.8.0"
     },
+    "require-dev": {
+        "nunomaduro/larastan": "^0.7.6",
```
Comment (yeganemehr) [inline composer.json]: Why did you need larastan?

File: src/Http/Controllers/GraphqlController.php [hunk 3]
```diff
@@ -16,21 +16,32 @@
...
 
-	public function __construct(Application $app)
+	public function __construct(IPackages $packages)
 	{
-		$this->app = $app;
+		$this->packages = $packages;
 	}
 
+	/**
+	 * @return \Illuminate\Http\Response|\Illuminate\Http\JsonResponse|\Laravel\Lumen\Http\ResponseFactory|mixed
```
Comment (yeganemehr) [inline src/Http/Controllers/GraphqlController.php]: this equals to (BS|BS|BS|true) that always is true

File: src/Http/Controllers/GraphqlController.php [hunk 4]
```diff
@@ -16,21 +16,32 @@
...
-		$builtin = Parser::parse(file_get_contents(package()->path("buildin-definations.graphql")));
+		$builtinFileContent = file_get_contents(package()->path("buildin-definations.graphql"));
+		if ($builtinFileContent === false) {
+			throw new \RuntimeException("can not read content of file 'buildin-definations.graphql'");
+		}
+		$builtin = Parser::parse($builtinFileContent);
 		$schema = BuildSchema::build($builtin);
-		foreach ($this->app->packages->all() as $package) {
+		foreach ($this->packages->all() as $package) {
+			/** @var \Jalno\Lumen\Contracts\IPackage&IGraphQLable $package */
```
Comment (yeganemehr) [inline src/Http/Controllers/GraphqlController.php]: phpstan can't understand it?

**Review by yeganemehr — CHANGES_REQUESTED** [review]

File: src/Http/Controllers/GraphqlController.php (hunk 4 above)
Comment (hosni) [reply src/Http/Controllers/GraphqlController.php]: No, it's thought the `$package` is JUST an instance of `IGraphQLable` and triggers an error that says: `call to undefined function path on $package`.\
I search for that and find the `instanceof` is the best and fastest operator for this type of operation and the solution is to use union types.

File: src/Http/Controllers/GraphqlController.php (hunk 3 above)
Comment (hosni) [reply src/Http/Controllers/GraphqlController.php]: I changed it to mixed

File: composer.json (hunk 2 above)
Comment (hosni) [reply composer.json]: I need `larastan` for this line:
https://github.com/Jalno/graphql/blob/0d7111a7d57708d0db78ebfe4c1789e254116208/src/Http/Controllers/GraphqlController.php#L106

The `phpstan` is not understood `response()` function.
